classes/ActionScheduler_ListTable.php

```diff
@@ -0,0 +1,517 @@
+			'singular' => __( 'action-scheduler', 'action-scheduler' ),
+			'plural'   => __( 'action-scheduler', 'action-scheduler' ),
+			'ajax'     => false,
+		) );
+	}
+
+	/**
+	 * Convert a interval of seconds into a two part human friendly string.
+	 *
+	 * The WordPress human_time_diff() function only calculates the time difference to one degree, meaning
+	 * even if an action is 1 day and 11 hours away, it will display "1 day". This funciton goes one step
+	 * further to display two degrees of accuracy.
+	 *
+	 * Inspired by the Crontrol::interval() function by Edward Dale: https://wordpress.org/plugins/wp-crontrol/
+	 *
+	 * @param int $interval A interval in seconds.
+	 * @return string A human friendly string representation of the interval.
+	 */
+	private static function human_interval( $interval, $periods_to_include = 2 ) {
+
+		if ( $interval <= 0 ) {
+			return __( 'Now!', 'action-scheduler' );
+		}
+
+		$output = '';
+
+		for ( $time_period_index = 0, $periods_in_interval = 0, $periods_included = 0, $seconds_remaining = $interval; $time_period_index < count( self::$time_periods ) && $seconds_remaining > 0 && $periods_included < $periods_to_include; $time_period_index++ ) {
+
+			$periods_in_interval = floor( $seconds_remaining / self::$time_periods[ $time_period_index ]['seconds'] );
+
+			if ( $periods_in_interval > 0 ) {
+				if ( ! empty( $output ) ) {
+					$output .= ' ';
+				}
+				$output .= sprintf( _n( self::$time_periods[ $time_period_index ]['names'][0], self::$time_periods[ $time_period_index ]['names'][1], $periods_in_interval, 'action-scheduler' ), $periods_in_interval );
+				$seconds_remaining -= $periods_in_interval * self::$time_periods[ $time_period_index ]['seconds'];
+				$periods_included++;
+			}
+		}
+
+		return $output;
+	}
+
+	/**
+	 * Returns the recurrence of an action or 'Non-repeating'. The output is human readable.
+	 *
+	 * @param ActionScheduler_Action $action
+	 *
+	 * @return string
+	 */
+	protected function get_recurrence( $action ) {
+		if ( $action->get_schedule()->is_recurring() ) {
+			return sprintf( __( 'Every %s', 'action-scheduler' ), self::human_interval( $recurrence->interval_in_seconds() ) );
+		}
+		return __( 'Non-repeating', 'action-scheduler' );
+	}
+
+	/**
+	 * Serializes the argument of an action to render it in a human friendly format.
+	 *
+	 * @param array $row The array representation of the current row of the table
+	 *
+	 * @return string
+	 */
+	public function column_args( array $row ) {
+		if ( empty( $row['args'] ) ) {
+			return;
+		}
+
+		$row_html = '<ul>';
+		foreach ( $row['args'] as $key => $value ) {
+			$row_html .= sprintf( '<li><code>%s => %s</li></code>', $key, $value );
+		}
+		$row_html .= '</ul>';
+
+		return $row_html;
+	}
+
+	/**
+	 * Prints the logs entries inline. We do so to avoid loading Javascript and other hacks to show it in a modal.
+	 *
+	 * @param array $row Action array.
+	 */
+	public function column_log_entries( array $row ) {
```

Comment:
@JPry why? Are you confusing it with `print_log_entry()`?

Reply:
Regarding this and [my other comment](https://github.com/Prospress/action-scheduler/pull/113#discussion_r174603300), this actually goes back to the `WP_List_Table` class itself. Inside `WP_List_Table::single_row_columns()`, these methods are called thanks to this `elseif` block (`*` added to indicate the appropriate line):

```php
} elseif ( method_exists( $this, 'column_' . $column_name ) ) {
	echo "<td $attributes>";
*	echo call_user_func( array( $this, 'column_' . $column_name ), $item );
	echo $this->handle_row_actions( $item, $column_name, $primary );
	echo "</td>";
}
```

The returned values of the method for each row are immediately echoed by the calling function. Also, the `ActionScheduler_ListTable::column_args()` method returns a string rather than calling `echo` directly. This is why I was saying that the generated strings should be returned instead of echoed.

As for the black magic of `PP_List_Table`, here's the order of calls that ultimately lead to each column being called:

* `PP_List_Table::display()`
* `WP_List_Table::display()`
* `WP_List_Table::display_rows_or_placeholder()`
* `WP_List_Table::display_rows()`
* `WP_List_Table::single_row()` (called within `foreach` loop for all rows)
* `WP_List_Table::single_row_columns()`
* `echo call_user_func( array( $this, 'column_' . $column_name ), $item );` (called within a `foreach` loop for with all columns)

Reply:
@JPry thanks for the explanation. That means some other `column_` methods need to be updated too, both in here and `PP_List_Table`. I'll get that fixed up now.